skimage/filters/_frangi.py

```diff
@@ -0,0 +1,180 @@
+
+        # Store the results in 3D matrices
+        filtered_list.append(filtered)
+
+    return filtered_list
+
+
+def frangi(image, scale=(1, 10), scale_step=2, beta1=0.5, beta2=15,
+                  black_ridges=True):
+    """Returns image filtered with Frangi filter.
+
+    Calculates the eigenvectors of the Hessian to compute the likeliness of
+    an image region to vessels, according to the method described in _[1].
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale : tuple of floats, optional
+        The range of sigmas used.
+    scale_step : float, optional
+        Step size between sigmas.
+    beta1 : float, optional
+        Frangi correction constant.
+    beta2 : float, optional
+        Frangi correction constant.
+    black_ridges : boolean, optional
+        Detect black ridges (default) set to true, for
+        white ridges set to false.
+
+    Returns
+    -------
+    out : (N, M) ndarray
+        Filtered image (maximum of pixels across all scales).
+
+    Notes
+    -----
+    Written by Marc Schrijver, 2/11/2001
+    Re-Written by D. J. Kroon University of Twente (May 2009)
+
+    References
+    ----------
+    .. [1] A. Frangi, W. Niessen, K. Vincken, and M. Viergever. "Multiscale
+           vessel enhancement filtering," In LNCS, vol. 1496, pages 130-137,
+           Germany, 1998. Springer-Verlag.
+    .. [2] Kroon, D.J.: Hessian based frangi vesselness filter.
+    .. [3] http://mplab.ucsd.edu/tutorials/gabor.pdf.
+    """
+
+    filtered_list = filtered_image_calculation(image, scale, scale_step, beta1, beta2)
+
+
+    # Return for every pixel the value of the scale(sigma) with the maximum
+    # output pixel value
+
+    return np.maximum.reduce(filtered_list)
+
+
+def hessian(image, scale=(1, 10), scale_step=2, beta1=0.5, beta2=15):
+    """Returns image filtered with hybrid hessian filter.
+
+    Almost equal to frangi filter, but uses alternative method of smoothing.
+    Can be used to detect edges.
+
+    Parameters
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale : tuple of floats, optional
+        The range of sigmas used.
+    scale_step : float, optional
+        Step size between sigmas.
+    beta1 : float, optional
+        Frangi correction constant.
+    beta2 : float, optional
+        Frangi correction constant.
+
+    Returns
+    -------
+    out : (N, M) ndarray
+        Filtered image (maximum of pixels across all scales).
+
+    Notes
+    -----
+    Written by Marc Schrijver, 2/11/2001
+    Re-Written by D. J. Kroon University of Twente (May 2009)
+
+    References
+    ----------
+    .. [1] Choon-Ching Ng, Moi Hoon Yap, Nicholas Costen and Baihua Li,
+           "Automatic Wrinkle Detection using Hybrid Hessian Filter".
+    """
+
+    frangi_ = False
+    filtered_list = filtered_image_calculation(image, scale, scale_step, beta1, beta2, frangi_)
```

Comment:
Also, Pythonic way would be using a keyword argument, `frangi_=False`, rather than a temporary variable. Having said that, this will be unnecessary once the function is refactored to not take the Frangi argument.